controller/src/controller.rs

```diff
@@ -271,34 +278,30 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
             let active_set_size = active_set.len();
             event!(Level::TRACE, ?active_set, "Found active set of nodes.");
 
-            // update time window: users can specify a update time window to operate Bottlerocket nodes update. If current time isn't within the time window,
-            // controller shouldn't have any action on it.
-            match within_time_window()? {
-                // If controller has already started a node update but the update time window stops,
-                // controller should respect it, continue to complete the update, and stop actions on remaining nodes.
-                // This logic will cooperate with 6 mins buffer strategy.
-                false => {
-                    // try to find out if any nodes are being updated. If yes, controller
-                    // will complete them and pause actions on other waitingForUpdate nodes.
-                    if !active_set.is_empty() {
-                        self.progress_active_set(active_set).await?;
-                    } else {
-                        sleep(ACTION_INTERVAL).await;
-                        continue;
-                    }
-                }
-                true => {
-                    if !active_set.is_empty() {
-                        self.progress_active_set(active_set).await?;
-                    }
+            // complete any ongoing updates even if we are outside of a scheduled window.
+            if !active_set.is_empty() {
+                self.progress_active_set(active_set).await?;
+            } else {
+                // If there are no more active nodes and current is outside of a scheduled window, brupop controller
+                // will sleep until next scheduled time.
+                brupop_scheduler
+                    .prepare_and_trigger_updates()
```

Comment:
nit: `wait_until_next_maintainence_window()` might be a more apt name, since `prepare` sounds as if there's more going on here.